packages/features/bookings/lib/useCheckOverlapWithOverlay.tsx

```diff
@@ -0,0 +1,44 @@
+import type { Dayjs } from "@calcom/dayjs";
+import dayjs from "@calcom/dayjs";
+
+import { useOverlayCalendarStore } from "../Booker/components/OverlayCalendar/store";
+
+function getCurrentTime(date: Date) {
+  const hours = date.getHours().toString().padStart(2, "0");
+  const minutes = date.getMinutes().toString().padStart(2, "0");
+  return `${hours}:${minutes}`;
+}
+
+export function useCheckOverlapWithOverlay(start: Dayjs) {
+  const overlayBusyDates = useOverlayCalendarStore((state) => state.overlayBusyDates);
+  let overlappingTimeStart: string | null = null;
+  let overlappingTimeEnd: string | null = null;
+
+  const isOverlapping =
+    overlayBusyDates &&
+    overlayBusyDates
+      .filter((value) => dayjs(value.start).isSame(start, "day"))
+      .some((busyDate) => {
+        const busyDateStart = new Date(busyDate.start);
+        const busyDateEnd = new Date(busyDate.end);
+        const startDate = start.toDate();
+
+        const overlap =
+          startDate >= busyDateStart &&
+          startDate <= busyDateEnd &&
+          dayjs(startDate).format("HH:mm") !== dayjs(busyDateEnd).format("HH:mm");
```

Comment:
If the HH:mm match that means we're at the end of a block 

slot 13:00 and busy time ends at 13:00  -> we dont need to class this as an overlap as they are free for 13:00